packages/konsole-app/src/hooks/useResources.ts

```diff
@@ -0,0 +1,20 @@
+//
+// Copyright 2021 DXOS.org
+//
+
+import { useEffect, useState } from 'react';
+
+import { IQuery, Resource } from '@dxos/registry-api';
+
+import { useRegistryClient } from './useRegistry';
+
+export function useResources (query?: IQuery): Resource[] {
+  const [resources, setResources] = useState<Resource[]>([]);
+  const registryClient = useRegistryClient();
+
+  useEffect(function () {
+    void registryClient.registry.getResources(query).then(setResources);
+  }, [query]);
+
+  return resources;
```

Comment:
Shouldn't this return something like `undefined` the first time it's called? Otherwise the user can't tell if resources are loading or if there are no resources.

Reply:
sure, thx